File: src/beginner/day04/Question02.java
package beginner.day04;

public class Question02 {

    public static class Solution {

        public int solution(int n) {
            int answer = 0;

            for (int i = 1; i <= 50; i++) {
                if ((i * 6) % n == 0) {
                    answer = i;
                    break;
                }
            }
            
            return answer;
        }
    }

    public static void main(String[] args) {

        Solution solution = new Solution();

        System.out.println(solution.solution(6));
        System.out.println(solution.solution(10));
        System.out.println(solution.solution(4));
    }
}

File: src/beginner/day20/Question02.kt
package beginner.day20

import kotlin.math.abs

class Question02{

    fun solution(keyinput: Array<String>, board: IntArray): IntArray {
        var answer: IntArray = intArrayOf(0, 0)

        var xLimit = board[0] / 2
        var yLimit = board[1] / 2

        for (input in keyinput) {
            when (input) {
                "left" -> if (abs(answer[0] - 1) <= xLimit) answer[0]--
                "right" -> if (abs(answer[0] + 1) <= xLimit) answer[0]++
                "up" -> if (abs(answer[1] + 1) <= yLimit) answer[1]++
                "down" -> if (abs(answer[1] - 1) <= yLimit) answer[1]--
            }
        }

        return answer
    }
}

fun main() {

    val solution = Question02()

    val keyinput1 = arrayOf("left", "right", "up", "right", "right")
    val board1 = intArrayOf(11, 11)

    val keyinput2 = arrayOf("down", "down", "down", "down", "down")
    val board2 = intArrayOf(7, 9)

    println(solution.solution(keyinput1, board1).contentToString())
    println(solution.solution(keyinput2, board2).contentToString())
}